pgml-admin/app/models.py

```diff
@@ -21,13 +20,32 @@ class Snapshot(models.Model):
     test_size = models.FloatField()
     test_sampling = models.TextField()
     status = models.TextField()
+    columns = models.JSONField(null=True)
+    analysis = models.JSONField(null=True)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
 
     class Meta:
         db_table = '"pgml"."snapshots"'
         managed = False
 
+    def sample(self, limit=1000):
+        with connection.cursor() as cursor:
+            cursor.execute(f"SELECT * FROM {self.relation_name} LIMIT %s", [limit])
```

Comment:
This will give you generally the same set when called multiple times due to caching/rows in shared buffers. If you want more random samples, not sure what the best way would be to go about it.

Reply:
That's what the order by random is for initially. It's OK, even desireable-ish that it's the same set on follow up reloads.